meilisearch-http/src/index/search.rs

```diff
@@ -515,11 +521,14 @@ impl<'a, A: AsRef<[u8]>> Formatter<'a, A> {
 
         tokens
             .map(|(word, token)| {
-                if need_to_highlight && token.is_word() && matcher.matches(token.text()) {
+                if need_to_highlight && token.is_word() && matcher.matches(token.text()).is_some() {
                     let mut new_word = String::new();
                     new_word.push_str(&self.marks.0);
-                    new_word.push_str(&word);
-                    new_word.push_str(&self.marks.1);
+                    if let Some(match_len) = matcher.matches(token.text()) {
+                        new_word.push_str(&word[..match_len]);
+                        new_word.push_str(&self.marks.1);
+                        new_word.push_str(&word[match_len..]);
+                    }
                     new_word
                 } else {
                     word.to_string()
```

Comment:
Here you could maybe use a [`Cow<str>`](https://doc.rust-lang.org/std/borrow/enum.Cow.html) instead of always creating a `String` as you collect in a `String` and [`String` implement `FromIterator<Item = Cow<str>>`](https://doc.rust-lang.org/std/string/struct.String.html#impl-Extend%3CCow%3C%27a%2C%20str%3E%3E). Doing so would help reduce the number of memory allocations we do.